tests/vmi_hostdev_test.go

```diff
@@ -1,76 +1,137 @@
 package tests_test
 
 import (
+	"context"
+	"fmt"
 	"strings"
+	"time"
 
 	expect "github.com/google/goexpect"
 	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
 	. "github.com/onsi/gomega"
 
+	"kubevirt.io/kubevirt/tests/flags"
 	"kubevirt.io/kubevirt/tests/util"
 
 	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
 	v1 "kubevirt.io/api/core/v1"
 	"kubevirt.io/client-go/kubecli"
 
+	k8sv1 "k8s.io/api/core/v1"
+
 	virtconfig "kubevirt.io/kubevirt/pkg/virt-config"
 	"kubevirt.io/kubevirt/tests"
 	"kubevirt.io/kubevirt/tests/console"
 )
 
+const (
+	failedDeleteVMI = "Failed to delete VMI"
+)
+
 var _ = Describe("[Serial][sig-compute]HostDevices", func() {
-	var err error
-	var virtClient kubecli.KubevirtClient
+	var (
+		err        error
+		virtClient kubecli.KubevirtClient
+		config     v1.KubeVirtConfiguration
+	)
+	restartVirtHandlerPods := func() {
+		listOptions := metav1.ListOptions{LabelSelector: v1.AppLabel + "=virt-handler"}
+		virtHandlerPods, err := virtClient.CoreV1().Pods(flags.KubeVirtInstallNamespace).List(context.TODO(), listOptions)
+		initialNumVirtHandler := len(virtHandlerPods.Items)
+		util.PanicOnError(err)
+		for _, pod := range virtHandlerPods.Items {
+			Eventually(func() error {
+				return virtClient.CoreV1().Pods(flags.KubeVirtInstallNamespace).Delete(context.Background(), pod.Name, metav1.DeleteOptions{})
+			}, 10*time.Second, 1*time.Second).Should(Succeed(), "Should delete virt-handler pod")
+
+		}
+		Eventually(func() bool {
+			// all virt-handler pods should have be restarted
+			virtHandlerPods, err := virtClient.CoreV1().Pods(flags.KubeVirtInstallNamespace).List(context.TODO(), listOptions)
+			util.PanicOnError(err)
+			if initialNumVirtHandler != len(virtHandlerPods.Items) {
+				return false
+			}
+			for _, pod := range virtHandlerPods.Items {
+				virtClient.CoreV1().Pods(flags.KubeVirtInstallNamespace).Get(context.Background(), pod.Name, metav1.GetOptions{})
+				util.PanicOnError(err)
+				if pod.Status.Phase != k8sv1.PodRunning {
+					return false
+				}
+			}
+			return true
+
+		}, 20*time.Second, 1*time.Second).Should(BeTrue())
+	}
 
 	BeforeEach(func() {
 		virtClient, err = kubecli.GetKubevirtClient()
 		util.PanicOnError(err)
+		kv := util.GetCurrentKv(virtClient)
+		config = kv.Spec.Configuration
+	})
+
+	AfterEach(func() {
+		kv := util.GetCurrentKv(virtClient)
+		// Reinitialized the DeveloperConfiguration to avoid to influence the next test
+		config = kv.Spec.Configuration
+		config.DeveloperConfiguration = &v1.DeveloperConfiguration{}
+		tests.UpdateKubeVirtConfigValueAndWait(config)
+		// TODO remove virt-handler pod restart
+		// We restart virt-handler because currently we don't support the device update of an already started device plugin
+		restartVirtHandlerPods()
```

Comment:
Update kubevirt is not enough. In my case, the second test was always failing. Maybe instead of restarting all the virt-handler pod I can wait unit the socket is cleaned up in `/var/lib/kubelet/device-plugins`. I'll give a try

Reply:
My bad! I was updating the DeveloperConfig and not the PermittedHostDevices :facepalm: